controllers/postgresqlhostcredentials_controller.go

```diff
@@ -0,0 +1,87 @@
+/*
+Copyright 2021.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/google/uuid"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	postgresqlv1alpha1 "go.lunarway.com/postgresql-controller/api/v1alpha1"
+)
+
+// PostgreSQLHostCredentialsReconciler reconciles a PostgreSQLHostCredentials object
+type PostgreSQLHostCredentialsReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+//+kubebuilder:rbac:groups=postgresql.lunar.tech,resources=postgresqlhostcredentials,verbs=get;list;watch;create;update;patch;delete
```

Comment:
I wouldn't expect the controller to manage these it self so I think we should reduce the scope of the roles here.

```suggestion
//+kubebuilder:rbac:groups=postgresql.lunar.tech,resources=postgresqlhostcredentials,verbs=get;list;watch
```

This change requires regenerating the manifests to make it change the generated k8s RBAC role.